conftest.py

```diff
@@ -25,10 +25,12 @@
 logger = logging.getLogger(__name__)
 
 
-def pytest_sessionstart():
+@pytest.fixture(scope="session", autouse=True)
+def testdata():
     """ Download sct_testing_data prior to test collection. """
     logger.info("Downloading sct test data")
-    downloader.main(['-d', 'sct_testing_data', '-o', sct_test_path()])
+    if not os.path.isdir(sct_test_path()):
```

Comment:
maybe .isdir() isn't reliable enough here?

we should probably be checking some expected test data version number somewhere here.

Reply:
Related to #2959. 

> * Implement some kind of caching mechanism to avoid re-downloading the same data.
>     * Should verify that data exists locally.
>     * Should verify that local data integrity is ok.
>     * This means that the data integrity should be published somewhere. Possibly in a manifest alongside the actual data. Alternatively on official website, github repo.

Reply:
Since we already have `test_data_integrity` and `check_test_data_integrity` in `conftest.py`, it seems simple enough to update https://github.com/sct-data/sct_testing_data with a manifest file, I think?

Reply:
This is probably out of scope of this PR, isn't it. :sweat_smile:

Reply:
Not necessarily, but ..I'd rather get this in before #3125 so that I don't need that `pytest -n 1` workaround there.

I think the point of `check_test_data_integrity` is to make sure the tests didn't do any in-place edits on the files. So, not exactly the same, but we could repurpose it. But yes, that should get rolled into our package manager! Maybe as a stop-gap we could put it into `sct_download_data` (it would help #3138 too!). And I don't know how distriploy is going to fit in there exactly.

Unfortunately, I don't think apt nor pip nor even brew have integrity checking. They can do *package* signing, optionally, but once a package is installed the only way to verify it's the right thing is to reinstall it. :exploding_head:

Reply:
I don't know. The `isdir()` check is super weak. The previous version just redownloaded the data every single time (and once per worker, even) -- you could see the multiple progress bars. Maybe we should *not* isdir() at all and move this discussion into #3138.

Reply:
> I don't know. The isdir() check is super weak. The previous version just redownloaded the data every single time (and once per worker, even) -- you could see the multiple progress bars. Maybe we should not isdir() at all and move this discussion into #3138.

I think I agree with removing the `isdir` check. Even if the data gets redownloaded every time, this is still an improvement, since you've removed the "once per worker" part that caused the concurrency issues.

Reply:
Alright done: https://github.com/neuropoly/spinalcordtoolbox/pull/3152/files#r554118620